### DISCUSSION TITLE
NPM/GitHub Account link "Act on your behalf" permission

### DISCUSSION THREAD

**Discussion by pcjmfranken** [npm/feedback#747] [Website] [answered]
The NPM/GitHub account link app requires the "act on your behalf" permission. What actions would the app take in GitHub on my behalf for its stated purpose of verifying my account? It seems unnecessary.

Pure curiosity - compounded by the fact that the app is closed source :wink:

**Comment by MylesBorins** [comment, marked as answer]
@pcjmfranken we take no actions on behalf of our users. The "act on your behalf" is applied to any GitHub App that has access to certain information, this is something we've raised to our identity team as giving the impression of over reach and not reflective of the actual permissions. I'll be sharing this thread with them to see if we can make changes to the GitHub App's themselves (not just our app) to make this less concerning

**Reply by pcjmfranken** [reply]
Thank you very much, I greatly appreciate this response!

I would like to add that personally I was not concerned that this would be abused by NPM. It was, as I said, pure curiosity.
